### ISSUE TITLE
About training epoch

### ISSUE THREAD

**Issue by jintaiWang** [pansanity666/TransHuman#2]
Hi! Thanks for your amazing work!
Recently, we retrained your code on thuman2 dataset.
For sanity checks, we train your code on ZJU-mocap, but we don't get meaningful images after 50 epochs. We just got some all-black images.

How many epochs do we need to get a meaningful output image during training? Could you provide more training details?
Thank you in advance.

**Comment by pansanity666** [comment]
All-black images usually mean collapse for NeRF, there must be something wrong.
Are you able to reproduce the inference results using our official checkpoint?

**Comment by pansanity666** [comment]
### About training details:
We provide two training schedulers in `train_or_eval.yaml`, one is `fast scheduler` (Line 91-98) which is used for fast validation, and another is `long scheduler` (Line 100-108) which is the final one used in the paper. 
      
+ `fast scheduler` do not use the LPIPS loss and neither the path sampling which trains for 500 epochs, actually, based on my experience, the best performance epoch is around epoch 275. It takes around half a day to train on 8xV100 (16GB) machine.
       
+ `long scheduler`  is the final version with LPIPS loss and path sampling used which trains for totally 3000 epochs, and the best performance epoch is around epoch 2100. It takes around 2 days to train on 8xV100 (32GB) machine.

NOTE: Remember to set the `use_patch_sampling` as `False` if you are using  `fast scheduler`. 

### About all-black images:
I have just verified the code with `fast scheduler` , and the visualization result after training for 25 epochs is:
     
![image](https://github.com/pansanity666/TransHuman/assets/47111102/5b87f4b4-b4ef-427e-813f-5c1b22b797dc)

, and the`img_loss` decresed to  around `0.007` afther 25 epochs training. 

Hope it helps you.

Best,

**Comment by jintaiWang** [comment]
Thanks for your reply!
We train your code on ZJU-Mocap but failed to get meaningful image(all-black) using one card setting.

**Comment by pansanity666** [comment]
+ If you are using 1 card for training, based on my experience, I recommend to adjust the 'random seed' and lower down the 'learning rate' in config file. Sometimes it works since the conditional NeRF is sensitive to initialization. 

+ Yesterday I have noticed the bug you mentioned about `human_name`. `human_name` is used in saving mesh, and I have forgotten to filter it in `trainer.py`. Adding
``` 
if k == 'human_name': 
     continue
```
to the `to_cuda` function in `trainer.py` will solve the bug.  I will push the fixed version later. 

Thanks for your interest.
Best,

**Comment by jintaiWang** [comment]
Thanks for your advise! 
It works when we set `lr = 7e-5` and `end_lr =1e-7` in config file.

**Comment by junjxiao** [comment]
> Hi! Thanks for your amazing work! Recently, we retrained your code on thuman2 dataset. For sanity checks, we train your code on ZJU-mocap, but we don't get meaningful images after 50 epochs. We just got some all-black images.
> 
> How many epochs do we need to get a meaningful output image during training? Could you provide more training details? Thank you in advance.

Hi, have you successfully retrained TransHuman on Thuman2.0 dataset? I have tried but got oom error on 2*4090 gpus.

**Comment by pansanity666** [comment]
Hi, Have you tried the ```fast scheduler``` mentioned above? It should work on 16GB GPUs.

**Comment by junjxiao** [comment]
> Hi, Have you tried the `fast scheduler` mentioned above? It should work on 16GB GPUs.

I have tried, but I don't know why it didn't work on 24g gpus. I use 4 input images as reference image, and I set  the parameter "rasterize" as False because I don't know what it means and how to compute it.

**Comment by junjxiao** [comment]
> Hi, Have you tried the `fast scheduler` mentioned above? It should work on 16GB GPUs.

One strange thing is that even I set num of sampled rays to 10 it's still OOM.

**Comment by junjxiao** [comment]
> Hi, Have you tried the `fast scheduler` mentioned above? It should work on 16GB GPUs.

I have found the problem, the size of input images in my code is set to 1024, I change it to 512 and it works.

**Comment by pansanity666** [comment]
> > Hi, Have you tried the `fast scheduler` mentioned above? It should work on 16GB GPUs.
> 
> I have found the problem, the size of input images in my code is set to 1024, I change it to 512 and it works.

Good!
